### PR TITLE
Ai is now independent from which side it's on.

### PR DESCRIPTION
**Pull request by PhiBabin**

### CODE REVIEW

File: Engine/controller.py [hunk 1]
```diff
@@ -76,6 +82,16 @@ def generate_packet(self, commands: Dict[int, Pose]) -> RobotState:
                             charge_kick=self[robot_id].engine_cmd.charge_kick))
         return packet
 
+    @staticmethod
+    def _put_in_robots_referiencial(robot, cmd):
```
Comment (sim0504) [inline Engine/controller.py]: typo

Comment (sim0504) [reply Engine/controller.py]: Bonne idée aussi!

File: Util/pose.py [hunk 2]
```diff
@@ -67,6 +67,9 @@ def to_array(self) -> np.ndarray:
     def to_dict(self) -> Dict[str, float]:
         return {'x': self.x, 'y': self.y, 'orientation': self.orientation}
 
+    def flip_x(self):
+        return Pose.from_values(-self.x, self.y, (2 * np.pi - self.orientation) % (2 * m.pi) - m.pi)
```
Comment (sim0504) [inline Util/pose.py]: make a wrap_to_pi helper function? avoid the confusion of seeing this line

Comment (sim0504) [reply Util/pose.py]: maybe mirror_x is a better name? not sure. flip_x seems to just flip the x component and hide the orientation. #veryneatpickyforgiveme

Comment (PhiBabin) [reply Util/pose.py]: The problem is that we already have a wrap_to_pi in geometry.py. But we can not import it since geometry depends on pose

Comment (PhiBabin) [reply Util/pose.py]: And we can not put it inside pose since it's also confusing.

Comment (PhiBabin) [reply Util/pose.py]: mirror_x fait bin plus de sense

Comment (sim0504) [reply Util/pose.py]: actually, it's the `import from` that cause problem. module import `import Pose` should avoid the circular import.

File: Util/pose.py [hunk 3]
```diff
@@ -68,7 +72,7 @@ def to_dict(self) -> Dict[str, float]:
         return {'x': self.x, 'y': self.y, 'orientation': self.orientation}
 
     def flip_x(self):
-        return Pose.from_values(-self.x, self.y, (2 * np.pi - self.orientation) % (2 * m.pi) - m.pi)
+        return Pose.from_values(-self.x, self.y, geometry.wrap_to_pi(self.orientation + np.pi))
```
Comment (PhiBabin) [inline Util/pose.py]: ` np.pi - self.orientation`

Comment (sim0504) [reply Util/pose.py]: es tu sur?

Comment (PhiBabin) [reply Util/pose.py]: C'est de même partout ailleurs et Stack Overflow est de mon bord.
Ex: 30deg sont miroir est 150deg.
180 + 30 = 210
180 - 30 = 150

**Pull request merged**